### PR TITLE
[tests-only][full-ci] tests: adding test coverage for thumbnail service

### PR DESCRIPTION
**Pull request by ghost**
## Description
When the image exceeded the any value in env
- THUMBNAILS_MAX_INPUT_IMAGE_FILE_SIZE
- THUMBNAILS_MAX_INPUT_WIDTH
- THUMBNAILS_MAX_INPUT_HEIGHT

the server would return status code `500` which shouldn't be the case as mentioned in [#10589](https://github.com/owncloud/ocis/issues/10589)

This PR adds test coverage for issue [#10589](https://github.com/owncloud/ocis/issues/10589). making sure the correct status code is returned.

Imp discussion : https://github.com/owncloud/ocis/pull/10763#discussion_r1891617222

## Related Issue

- related issue: https://github.com/owncloud/ocis/issues/10725
- bug report: https://github.com/owncloud/ocis/issues/10589

## How Has This Been Tested?

- test environment:
- test case 1: locally
- test case 2: ci
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 1]
```diff
@@ -367,3 +367,43 @@ Feature: previews of files downloaded through the webdav API
       | spaces           | fill      | fill.png       |
       | spaces           | resize    | resize.png     |
       | spaces           | thumbnail | thumbnail.png  |
+
+  @issue-10589 @env-config
+  Scenario Outline: preview thumbnail of an image with diffrent envs
```
Comment (prashant-gurung899) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:372]:
```suggestion
  Scenario Outline: preview thumbnail of an image with different envs
```

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 2]
```diff
@@ -367,3 +367,43 @@ Feature: previews of files downloaded through the webdav API
...
+      | spaces           | THUMBNAILS_MAX_INPUT_IMAGE_FILE_SIZE | 1KB   |
+      | old              | THUMBNAILS_MAX_INPUT_WIDTH           | 200   |
+      | new              | THUMBNAILS_MAX_INPUT_WIDTH           | 200   |
+      | spaces           | THUMBNAILS_MAX_INPUT_WIDTH           | 200   |
+      | old              | THUMBNAILS_MAX_INPUT_HEIGHT          | 200   |
+      | new              | THUMBNAILS_MAX_INPUT_HEIGHT          | 200   |
+      | spaces           | THUMBNAILS_MAX_INPUT_HEIGHT          | 200   |
+
+  @issue-10589 @env-config
+  Scenario Outline: preview thumbnail of a file with diffrent envs
```
Comment (prashant-gurung899) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:392]:
```suggestion
  Scenario Outline: download thumbnail preview of a file with different envs
```

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature (hunk 1 above)
Comment (amrita-shrestha) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:372]:
```suggestion
  Scenario Outline: download previews of an image with different thumbnail related enviroment
```

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature (hunk 2 above)
Comment (amrita-shrestha) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:392]:
```suggestion
  Scenario Outline: download preview of a file with different thumbnail related enviroment
```

**Review by amrita-shrestha — APPROVED** [review]
lgtm

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 3]
```diff
@@ -367,3 +367,43 @@ Feature: previews of files downloaded through the webdav API
...
+    And the value of the item "/d:error/s:message" in the response about user "Brian" should be "thumbnails: image is too large"
+    Examples:
+      | dav-path-version | env                                  | value |
+      | old              | THUMBNAILS_MAX_INPUT_IMAGE_FILE_SIZE | 1KB   |
+      | new              | THUMBNAILS_MAX_INPUT_IMAGE_FILE_SIZE | 1KB   |
+      | spaces           | THUMBNAILS_MAX_INPUT_IMAGE_FILE_SIZE | 1KB   |
+      | old              | THUMBNAILS_MAX_INPUT_WIDTH           | 200   |
+      | new              | THUMBNAILS_MAX_INPUT_WIDTH           | 200   |
+      | spaces           | THUMBNAILS_MAX_INPUT_WIDTH           | 200   |
+      | old              | THUMBNAILS_MAX_INPUT_HEIGHT          | 200   |
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:387]: I would recommend to test them in one go using this step:
```feature
And the following configs have been set:
```

**Review by PrajwolAmatya — COMMENTED** [review]
LGTM

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 4]
```diff
@@ -367,3 +367,39 @@ Feature: previews of files downloaded through the webdav API
...
+  Scenario Outline: try to download a file preview when the maximum thumbnail input value in the environment is set to a small value
+    Given the following configs have been set:
+      | config                               | value |
+      | THUMBNAILS_MAX_INPUT_IMAGE_FILE_SIZE | 1KB   |
+      | THUMBNAILS_MAX_INPUT_WIDTH           | 200   |
+      | THUMBNAILS_MAX_INPUT_HEIGHT          | 200   |
+    And using <dav-path-version> DAV path
+    And user "Alice" has uploaded file "filesForUpload/lorem-big.txt" to "/lorem-big.txt"
+    When user "Alice" downloads the preview of "/lorem-big.txt" with width "36" and height "36" and processor thumbnail using the WebDAV API
+    Then the HTTP status code should be "403"
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:399]: what happens if we do not set the configs? 200?

Comment (saw-jan) [reply tests/acceptance/features/coreApiWebdavPreviews/previews.feature:399]: Good. It is understandable for image file but for a text file, how does the thumbnail service react? if the text file is under max input size?

Comment (saw-jan) [reply tests/acceptance/features/coreApiWebdavPreviews/previews.feature:399]: then the scenarios look good :+1:

**Review by saw-jan — APPROVED** [review]

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 5]
```diff
@@ -367,3 +367,39 @@ Feature: previews of files downloaded through the webdav API
...
+    And using <dav-path-version> DAV path
+    And user "Alice" has uploaded file "filesForUpload/testavatar.jpg" to "/testimage.jpg"
+    When user "Alice" downloads the preview of "/testimage.jpg" with width "32" and height "32" and processor thumbnail using the WebDAV API
+    Then the HTTP status code should be "403"
+    And the value of the item "/d:error/s:message" in the response about user "Brian" should be "thumbnails: image is too large"
+    Examples:
+      | dav-path-version |
+      | old              |
+      | new              |
+      | spaces           |
```
Comment (amrita-shrestha) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:387]: @nirajacharya2 this scenario give 200

Comment (amrita-shrestha) [reply tests/acceptance/features/coreApiWebdavPreviews/previews.feature:387]: your CI build is failing. Please adjust it according to your comment 
Screenshot from 2024-12-24 16-39-18

**Review by amrita-shrestha — APPROVED** [review]
lgtm

**Pull request merged**